drivers/serial/uart_nrfx_uarte.c

```diff
@@ -921,6 +998,11 @@ static int uarte_nrfx_rx_enable(const struct device *dev, uint8_t *buf,
 				async_rx->flush_cnt -= cpy_len;
 				memmove(cfg->rx_flush_buf, &cfg->rx_flush_buf[cpy_len],
 						async_rx->flush_cnt);
+				if (IS_ENABLED(UARTE_ANY_CACHE) &&
+				    (cfg->flags & UARTE_CFG_FLAG_CACHEABLE)) {
+					sys_cache_data_flush_range(cfg->rx_flush_buf,
```

Comment:
as above - `dmm` shouldn't be used here as this is a shim-specific code maintaining `rx_flush_buf` that happens to be located in cached RAM, right ?